Sources/AutobahnDescription/AutobahnDescription.swift

```diff
@@ -60,37 +62,41 @@ public final class Autobahn<H: Highway> {
     
     // MARK: - Public Functions
     
-    public init(_ highwayType: H.Type) {}
+    public init(_ highwayType: H.Type) {
+        self.highwayStart = Date()
+    }
     
     public func beforeAll(handler: @escaping (H) throws -> Void) -> Autobahn<H> {
-        guard self.beforeAllHandler == nil else {
-            fatalError("beforeAll declared more than once")
-        }
-        self.beforeAllHandler = handler
+        precondition(beforeAllHandler == nil, "beforeAll declared more than once")
+        beforeAllHandler = handler
         return self
     }
     
-    public func highway(_ highway: H, dependsOn: [H]? = nil, taskHandler: @escaping () throws -> Void) -> Autobahn<H> {
-        self.highways[highway] = taskHandler
-        if let d = dependsOn {
-            self.dependings[highway] = d
-        }
+    public func highway(_ highway: H, handler: @escaping () throws -> Void) -> Autobahn<H> {
+        self.highways[highway] = handler
+        return self
+    }
+
+    public func highway(_ highway: H, dependsOn highways: [H]) -> Autobahn<H> {
+        dependings[highway] = highways
+        return self
+    }
+
+    public func highway(_ highway: H, dependsOn highways: [H], handler: @escaping () throws -> Void) -> Autobahn<H> {
+        self.highways[highway] = handler
+        dependings[highway] = highways
         return self
     }
     
     public func afterAll(handler: @escaping (H) throws -> Void) -> Autobahn<H> {
-        guard self.afterAllHandler == nil else {
-            fatalError("afterAll declared more than once")
-        }
-        self.afterAllHandler = handler
+        precondition(afterAllHandler == nil, "afterAll declared more than once")
+        afterAllHandler = handler
         return self
     }
     
     public func onError(handler: @escaping (String, Error) -> Void) -> Autobahn<H> {
-        guard self.onErrorHandler == nil else {
-            fatalError("onError declared more than once")
-        }
-        self.onErrorHandler = handler
+        precondition(onErrorHandler == nil, "onError declared more than once")
```

Comment:
Same as `afterAll` handler, we could also make it possible to define multiple ones

Reply:
See above comment ^^